doc/neps/scope.rst

```diff
@@ -0,0 +1,41 @@
+==============
+Scope of NumPy
+==============
+
+Here, we describe aspects of N-d array computation that are within scope for NumPy development. This is *not* an aspirational definition of where NumPy should aim, but instead captures the status quo—areas which we have decided to continue supporting, at least for the time being.
+
+- **In-memory, N-dimensional, homogeneously typed (single pointer + strided) arrays on CPUs**
+
+  - Support for a wide range of data types
+  - Not specialized hardware such as GPUs
+  - But, do support wide range of CPUs (e.g. ARM, PowerX)
+
+- **Higher level APIs for N-dimensional arrays**
+
+  - NumPy is a *de facto* standard for array APIs in Python
+  - Indexing and fast iteration over elements (ufunc)
+  - Interoperability protocols with other data container implementations (like `__array_ufunc__`).
+
+- **Python API and a C API** to the ndarray's methods and attributes.
+
+- Other **specialized types of N-dimensional arrays**:
+
+  - Masked arrays
+  - Structured arrays
```

Comment:
Stuctured arrays at least partially fall under `dtype`; did you mean `recarray` here? Or just the interface for arrays to the dtype fields?

Reply:
We debated this back and forth, it's not really clear. There's more to structured arrays than dtypes, and recarray is mainly a thin interface to construct structured arrays (at least that's what my impression is). Structured arrays are the more important/fundamental thing I'd think.

The docs are very confused, the user guide only talk about structured arrays ([here](https://docs.scipy.org/doc/numpy/user/basics.rec.html)), while the reference guide only contains recarray ([here](https://www.numpy.org/devdocs/reference/generated/numpy.recarray.html?highlight=recarray#numpy.recarray)).

Reply:
In my mind, recarrays should probably not exist (they only provide attribute access to fields), and whatever functionality we need should instead be provided as structured arrays.  But utilities for loading / constructing such arrays (`np.lib.recfromcsv`, `np.lib.recfromtxt`) remain in scope?

Reply:
Agree with that sense: `recarray` adds very little to structured arrays. If we add `memmap` as well, then I think this list becomes somewhat logical, as it covers adding an array-like attribute, more complicated elementary structures, and a different back-end - all those are things that subclasses/duck arrays outside of numpy are providing as well.

Maybe the title of the section should be `Other specialized types or uses of N-dimensional arrays`?

Reply:
> The docs are very confused

IIRC, the name "structured arrays" was a late comer to the terminology and that might account for the confusing mix.